### PR TITLE
Testing cacheResources()

### PR DESCRIPTION
**Pull request by delapuente**
@marco-c do you mind to take a look? It's not complete but if you give me feedback now, I can progressively add more tests (a function per commit) and I can fix style errors ASAP.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ gulp.task('test-sw', function () {
 });
 
 gulp.task('pre-test', function () {
-  return gulp.src(['lib/**/*.js'])
+  return gulp.src(['lib/**/*.js,!lib/service-worker/**'])
```
Comment (marco-c) [inline gulpfile.js]: This path doesn't exist, does it?

Comment (marco-c) [inline gulpfile.js]: Maybe you haven't `git add`ed it?

Comment (delapuente) [inline gulpfile.js]: That's not even correct. Let me see what happen there.

**Comment by delapuente** [comment]
@marco-c I finished with the tests. All the API is covered. If you don't mind to take a second look. :)

**Comment by delapuente** [comment]
Fix #213

File: test/service-worker/testServiceWorker.js [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+  describe('cacheResources()', function () {
+
+    var mockedOkResponse = { ok: true };
+    var mockedFailingResponse = { ok: false };
+    var testTimestamp = 1234567890;
+
+    function allResourcesInCache(resources, cache) {
+      var requests = [];
+      var responses = [];
+      for (var call = 0, entry; entry = cache.put.args[call]; call++) {
```
Comment (marco-c) [inline test/service-worker/testServiceWorker.js]: Nit: forEach

File: test/service-worker/testServiceWorker.js [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+        assert(sameRequestURL(resultUrl, originalUrl));
+      });
+    });
+  });
+
+  describe('prepareCache()', function () {
+    function createNonEmptyCache(name) {
+      return self.caches.open(name)
+        .then(function (cache) {
```
Comment (marco-c) [inline test/service-worker/testServiceWorker.js]: Nit: for promises, we're using the Q-style in most of the code, e.g.:

``` Javascript
  return self.caches.open(name)
  .then(function(cache) {
    // ...
  });
```

It doesn't really matter though.

**Comment by marco-c** [comment]
Looks good to me.
What do you think about splitting the testServiceWorker.js file in multiple files? (one for each function)

**Comment by delapuente** [comment]
I will fix all the nits and I agree on splitting the test. Maybe not one per method but some of them would be more readable if they were in their own files.

**Pull request merged**